gdmlGeneratorV1_materials.pl

```diff
@@ -437,202 +476,342 @@
 
 ##-------------------------------------------------------------------------------##
 
-
 ##--------------------Start defining solids.-------------------------------------##
 
-open(def, ">", "solids${opt_T}.xml") or die "cannot open > solids${opt_T}.xml: $!";
+open( def, ">", "solids${opt_T}.xml" )
+  or die "cannot open > solids${opt_T}.xml: $!";
 print def "<solids>\n\n";
 
-#---------------------Defining solid of mother volume.-------------------------------------------------------------------------------------# 
-{ my $dxMabs; my $dxM2abs; my $dxMshift;
-$dxMabs = abs($dxMext);
-$dxM2abs = $dxM+2*$dxMabs;
-$dxMshift = ($dxMext<=>0)*($dxM+$dxMext)/2;
-
-print def "<box lunit=\"mm\" name=\"boxMotherSolBase${opt_T}\" x=\"$dxM\" y=\"$dyM\" z=\"$dzM\"/>\n";
-print def "<box lunit=\"mm\" name=\"boxMotherSolExt${opt_T}\" x=\"$dxMabs\" y=\"$dyM\" z=\"$dzM\"/>\n";
-
-print def "<union name =\"boxMotherSol${opt_T}\">
+#---------------------Defining solid of mother volume.-------------------------------------------------------------------------------------#
+{
+    my $dxMabs;
+    my $dxM2abs;
+    my $dxMshift;
+    $dxMabs   = abs($dxMext);
+    $dxM2abs  = $dxM + 2 * $dxMabs;
+    $dxMshift = ( $dxMext <=> 0 ) * ( $dxM + $dxMext ) / 2;
+
+    print def
+"<box lunit=\"mm\" name=\"boxMotherSolBase${opt_T}\" x=\"$dxM\" y=\"$dyM\" z=\"$dzM\"/>\n";
+    print def
+"<box lunit=\"mm\" name=\"boxMotherSolExt${opt_T}\" x=\"$dxMabs\" y=\"$dyM\" z=\"$dzM\"/>\n";
+
+    print def "<union name =\"boxMotherSol${opt_T}\">
 	<first ref=\"boxMotherSolBase${opt_T}\"/> 
 	<second ref=\"boxMotherSolExt${opt_T}\"/> 
 	<position unit=\"mm\" name=\"boxMotherPos${opt_T}\" x=\"$dxMshift\" y=\"0\" z=\"0\"\/>
 </union>\n\n";
 
-print def "<cone aunit=\"rad\" deltaphi=\"2*PI\" lunit=\"mm\" name=\"coneMotherSol${opt_T}\" rmax1=\"$drMinM\" rmax2=\"$drMinM\" rmin1=\"0\" rmin2=\"0\" startphi=\"0\" z=\"$dxM2abs\"/>\n";
+    print def
+"<cone aunit=\"rad\" deltaphi=\"2*pi\" lunit=\"mm\" name=\"coneMotherSol${opt_T}\" rmax1=\"$drMinM\" rmax2=\"$drMinM\" rmin1=\"0\" rmin2=\"0\" startphi=\"0\" z=\"$dxM2abs\"/>\n";
 
-print def "<subtraction name =\"logicMotherSol${opt_T}\">
+    print def "<subtraction name =\"logicMotherSol${opt_T}\">
 	<first ref=\"boxMotherSol${opt_T}\"/> 
 	<second ref=\"coneMotherSol${opt_T}\"/> 
 	<position unit=\"mm\" name=\"coneMotherPos${opt_T}\" x=\"0\" y=\"0\" z=\"0\"\/>
-        <rotation unit=\"rad\" name=\"coneMotherRot${opt_T}\" x=\"0\" y=\"PI/2\" z=\"0\"/>
+        <rotation unit=\"rad\" name=\"coneMotherRot${opt_T}\" x=\"0\" y=\"pi/2\" z=\"0\"/>
```

Comment:
Should all of these PI's be lowercase now?

Reply:
I don't know. Doesn't it work?

Reply:
Yes, it is working